Jsoup follow-up formatting patches

hazendaz opened the pull request:
- Use XXX rather than TODO as entries are not really todo's but need called out if upstream fixes
- Pull up line ending normalization as jsoup will cause non normalized data during its process even if file was previously normalized.  This was mixed into another fix.
- Add clarifications within index counter fix.  Issue typically shows up on span, div, a, possibly others where it still adds one extra space in violations to what was asked for index size
- Removed header line break fix as that was fixed in the fork we used and now also fixed in jsoup directly
- Add new line to end of the file

The existing checks were all verified to still be considerations that have not yet been fixed in jsoup pretty print.  There are 4 patches as follows:
- Removal of trailing space on each line
- Normalization of line endings (jsoup processing internally is LF which will cause mixed results in windows)
- Index counter problem resulting in some lines getting 1 extra space typically
- Add new line to end of file if missing

ctubbsii commented on src/main/java/net/revelc/code/formatter/jsoup/JsoupBasedFormatter.java:
```diff
@@ -71,36 +71,45 @@ public String doFormat(String code, final LineEnding ending) {
+            int trimLineLength;
             int remainder;
-            String[] lines = formattedCode.split("\\r?\\n");
+            lines = formattedCode.split(ending.getChars());
```
This line's change will leave behind extra chars if the detected line endings aren't consistent throughout the file (like if there's a mix of line ending styles in the file). The current code will split on any line, regardless of the line ending style.

hazendaz replied:
Added additional commits to help walk through changes on jsoup post pretty print processing.  Specific issue was done earlier in process to get it out of mixed patch so that each patch stands on its own.

hazendaz commented:
The splitter on mixed content was moved above this section so it only needs to look at mixed content the one time.  The subsequent ones have line ending then normalized.  See line 80.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Christopher Tubbs ***@***.***>
Sent: Wednesday, July 20, 2022 3:38:05 PM
To: revelc/formatter-maven-plugin ***@***.***>
Cc: Jeremy Landis ***@***.***>; Assign ***@***.***>
Subject: Re: [revelc/formatter-maven-plugin] Jsoup follow-up formatting patches (PR #645)


@ctubbsii commented on this pull request.

________________________________

In src/main/java/net/revelc/code/formatter/jsoup/JsoupBasedFormatter.java<https://github.com/revelc/formatter-maven-plugin/pull/645#discussion_r925988085>:

>              int remainder;
-            String[] lines = formattedCode.split("\\r?\\n");
+            lines = formattedCode.split(ending.getChars());


This line's change will leave behind extra chars if the detected line endings aren't consistent throughout the file (like if there's a mix of line ending styles in the file). The current code will split on any line, regardless of the line ending style.

—
Reply to this email directly, view it on GitHub<https://github.com/revelc/formatter-maven-plugin/pull/645#pullrequestreview-1045580680>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAHODI277P35CY2KDWWYYKLVVBIR3ANCNFSM53Y6GPMA>.
You are receiving this because you were assigned.Message ID: ***@***.***>

hazendaz commented:
Going to go ahead and merge as concern addressed by adding more clarity to the jsoup patches and our build is fully testing it with 2 passes on file.
